### ISSUE TITLE
torch.nn.functional.pad generates incomplete ONNX without explicit padding value

### ISSUE THREAD

**Issue by Hyunseok-Kim0** [pytorch/pytorch#84979]
### 🐛 Describe the bug

When I tried to export pytorch model which includes torch.nn.functional.pad to ONNX, it works without any problem.
Converted model works correctly and output is same with original pytorch model. 
The model without explicit padding value and the model with padding value 0 is shown below, respectively. 
![padding_problem](https://user-images.githubusercontent.com/34959032/190045012-e33d447f-f633-449a-a19c-9870de309098.png)
The official document (https://pytorch.org/docs/stable/generated/torch.nn.functional.pad.html) say torch.nn.functional.pad has default padding value of 0. However, the default padding value does not included in ONNX. 
This incomplete ONNX can occur problems when the model is converted to other formats like OpenVINO, TFlite, etc.

```python
import torch
import onnx
from onnxsim import simplify

class simple_padding(nn.Module):

    def __init__(self):
        super().__init__()

    def forward(self, x):

        pad_shape = [1, 1, 1, 1]
   
        # return torch.nn.functional.pad(x, pad_shape)
        return torch.nn.functional.pad(x, pad_shape, value=0)

def padding_export_check():

    onnx_save_path = "padding.onnx"
    simplified_save_path = "opt_paddingonnx"

    model = simple_padding()

    dummy_input = torch.rand(3, 224, 224, device='cpu')

    torch.onnx.export(model,
                      args=(dummy_input),
                      f=onnx_save_path,
                      input_names=["input"],
                      opset_version=11)

    # onnx simplify
    saved_onnx = onnx.load(onnx_save_path)
    onnx_simplified, check = simplify(saved_onnx)
    assert check, "Simplified ONNX model could not be validated"
    onnx.save(onnx.shape_inference.infer_shapes(onnx_simplified), simplified_save_path)
```

### Versions

Collecting environment information...
PyTorch version: 1.12.0
Is debug build: False
CUDA used to build PyTorch: 11.3
ROCM used to build PyTorch: N/A

OS: Ubuntu 18.04.6 LTS (x86_64)
GCC version: (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
Clang version: Could not collect
CMake version: Could not collect
Libc version: glibc-2.17

Python version: 3.7.13 (default, Mar 29 2022, 02:18:16)  [GCC 7.5.0] (64-bit runtime)
Python platform: Linux-5.4.0-125-generic-x86_64-with-debian-buster-sid
Is CUDA available: True
CUDA runtime version: Could not collect
GPU models and configuration: 
GPU 0: NVIDIA GeForce RTX 3090
GPU 1: NVIDIA GeForce RTX 3090
GPU 2: NVIDIA GeForce RTX 3090
GPU 3: NVIDIA GeForce RTX 3090

Nvidia driver version: 470.141.03
cuDNN version: Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.8.2.0
/usr/lib/x86_64-linux-gnu/libcudnn_adv_infer.so.8.2.0
/usr/lib/x86_64-linux-gnu/libcudnn_adv_train.so.8.2.0
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_infer.so.8.2.0
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_train.so.8.2.0
/usr/lib/x86_64-linux-gnu/libcudnn_ops_infer.so.8.2.0
/usr/lib/x86_64-linux-gnu/libcudnn_ops_train.so.8.2.0
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] adabelief-pytorch==0.2.1
[pip3] MoViNet-pytorch==0.0.1
[pip3] numpy==1.21.5
[pip3] torch==1.12.0
[pip3] torchelastic==0.2.0
[pip3] torchinfo==1.7.0
[pip3] torchmetrics==0.9.3
[pip3] torchsummary==1.5.1
[pip3] torchtext==0.13.0
[pip3] torchvision==0.13.0
[conda] Could not collect

**Comment by justinchuby** [comment]
Thanks for reporting this issue @Hyunseok-Kim0 ! We will look into this and update here.

**Comment by justinchuby** [comment]
@AllenTiTaiWang i see you were looking at the padding code. Do you have any suggestions or thoughts here?

**Comment by AllenTiTaiWang** [comment]
Sure, I will take a look.

**Comment by AllenTiTaiWang** [comment]
It seems to me that the current behavior actually makes more sense than change it. 
From ONNX perspective, based on ONNX spec - [Pad](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#Pad-11), `constant_value` is an optional value that could be blank. And in converter, [torch.nn.functional.pad](https://pytorch.org/docs/stable/generated/torch.nn.functional.pad.html) is also saying that `value` is optional, so I think it's reasonable that the converted graphs (with and without specifying value=0) are different. 

@justinchuby, what do you think?

**Comment by Hyunseok-Kim0** [comment]
I think the main problem is the inconsistency of `torch.onnx.export`. 
In torch 1.10 and 1.11, `torch.onnx.export` record default value 0 to `torch.nn.functional.pad` when user didn't specify padding value. Also the default initializer is included in onnx graph. 
However, torch 1.12 records nothing, thus user get `None` when trying to get initializer of onnx pad node from graph.
Even though padding value is optional, documented default padding mode is `constant` with default value 0. I think default initializer should be included.

**Comment by AllenTiTaiWang** [comment]
The default initializer says [`value` is None](https://pytorch.org/docs/stable/generated/torch.nn.functional.pad.html) actually since torch 1.12. PyTorch seems made the change in this [PR](https://github.com/pytorch/pytorch/pull/73431), and [documentation](https://github.com/pytorch/pytorch/pull/73433). That is, ONNX converter never changes our code on pad and is simply using their default to convert the model which I believe that we are following the upstream (torch) value. We could add something like _if we get None from `value` and set it to 0_, but that would be changing the behavior from upstream.

**Comment by abock** [comment]
Closing per @AllenTiTaiWang's previous comment.
